docs/enhancements/1-crds-and-controllers.md

```diff
@@ -0,0 +1,430 @@
+        - ratings
+  # This rule imports everything from the central cluster.
+  - meshID: central
+```
+
+Rules based on label selectors are useful when an admin wants to give users the control on exporting Services,
+while `serviceList` gives the admin full control on exported services.
+
+We are able to provide the same functionality with only label selectors for services and namespaces.
+In this variant, `export` is a list of rule sets containing `serviceSelector` or `namespaceSelector`.
+Rules are OR-ed, but each rule set (an item in the `export` list) is AND-ed.
+
+```yaml
+apiGroup: federation.openshift-service-mesh.io/v1alpha1
+kind: FederationServiceRules
+metadata:
+  # Name must be default.
+  name: default
+  # Namespace is expected to be the same as the controller's namespace.
+  namespace: istio-system
+spec:
+  # The following list of rules exports services "ratings" and "reviews" from namespace "ns-1" and service "details" from namespace "ns-2".
+  export:
+    selectors:
+    - serviceSelector:
+        matchExpressions:
+        - key: app.kubernetes.io/name
+          operator: In
+          values:
+          - ratings
+          - reviews
+      namespaceSelector:
+        matchLabels:
+          kubernetes.io/metadata.name: ns-1
+    - serviceSelector:
+        matchLabels:
+          app.kubernetes.io/name: details
+      namespaceSelector:
+        matchLabels:
+          kubernetes.io/metadata.name: ns-2
+    dnsSettings:
+      searchDomain: mesh.global
+  # Optional
+  # Empty import allows to not subscribe FDS.
+  import:
+  - meshID: west
+    # Optional
+    # If no selector is specified, then everything is imported.
+    serviceSelectors:
+    - matchExpressions:
+      - key: app.kubernetes.io/name
+        operator: In
+        values:
+        - ratings
+```
+
+Responsibilities of the controller for this resource:
+1. Owning resources for exported services:
+   * `Gateway`;
+   * `ServiceEntry` - if custom search domain configured;
+   * `EnvoyFilter` - if the ingress type is "openshift-router".
+2. Watching services and namespaces and check if they match export rules.
+3. Reconciling owned resource if `MeshFederation` was changed.
+
+Example controller:
+```go
+func (r *FederationServiceRulesReconciler) SetupWithManager(mgr ctrl.Manager) error {
+        return ctrl.NewControllerManagedBy(mgr).
+                For(&v1alpha1.FederationServiceRules{}).
+                Owns(&v1.Gateway{}).
+                Owns(&v1.ServiceEntry{}).
+                Owns(&v1.EnvoyFilter{}).
+                Watches(&corev1.Service{}, WithPredicates(checkIfMatchesExportRules)).
+                Watches(&corev1.Namespace{}, WithPredicates(checkIfMatchesExportRules)).
+                Watches(&MeshFederation{}).
+                Complete(r)
+}
+```
+
+All export-related resource will include owner reference pointing to `FederationServiceRules`,
+so deleting `FederationServiceRules` will result in removing these resources as well.
```

Comment:
I am afraid we are focusing too much on `ownerReference`s on its own, whereas the actual goal is to have a way of properly cleaning up related resources when given CR no longer exists. How it's done will emerge from the implementation.